src/core/freelist.lua

```diff
@@ -3,10 +3,10 @@ module(...,package.seeall)
 local ffi = require("ffi")
 
 function new (type, size)
-   return { nfree = 0,
-            max = size,
-            -- XXX Better LuaJIT idiom for specifying the array type?
-            list = ffi.new(type.."[?]", size) }
+   local element_ct = ffi.typeof(type)
+   local fl_type = ffi.typeof("struct { int nfree, max; $ list[?]; }",
```

Comment:
Why ints for `free` and `max`? Should it be a 64bit type?
What about cache alignment? I'd personally prefer `list` to start on an aligned address.

Did you measure the performance impact?